log10/bigquery.py

```diff
@@ -0,0 +1,67 @@
+from google.cloud import bigquery
+from google.api_core.exceptions import NotFound
+import os
+
+
+# todo: add requirements.txt file
+# todo: add instructions for bigquery integration
+
+def initialize_bigquery():
+
+    # Configure the BigQuery client
+    project_id = os.environ.get("PROJECT_ID")
+    dataset_id = os.environ.get("DATASET_ID")
+    table_id = os.environ.get("TABLE_ID")
+
+    client = bigquery.Client(project=project_id)
+
+    def dataset_exists(dataset_id):
+        try:
+            client.get_dataset(dataset_id)  # API request
+            return True
+        except NotFound:
+            return False
+
+    def table_exists(dataset_id, table_id):
+        try:
+            table_ref = client.dataset(dataset_id).table(table_id)
+            client.get_table(table_ref)  # API request
+            return True
+        except NotFound:
+            return False
+
+    # Check if dataset exists
+    if dataset_exists(dataset_id):
+        print(f"Dataset {dataset_id} exists.")
+    else:
+        print(f"Dataset {dataset_id} does not exist. Creating...")
+        # Create the dataset
+        dataset_ref = client.dataset(dataset_id)
+        dataset = bigquery.Dataset(dataset_ref)
+        # Set the location, e.g., "US", "EU", "asia-northeast1", etc.
+        dataset.location = "US"
+        created_dataset = client.create_dataset(dataset)  # API request
+        print(f"Dataset {created_dataset.dataset_id} created.")
+
+    client_dataset = client.dataset(dataset_id)
+
+    # Check if table exists
+    if table_exists(dataset_id, table_id):
+        print(f"Table {table_id} exists in dataset {dataset_id}.")
```

Comment:
We may have to some backend code to initialize the big query tables. Having that done by client code may be a bit surprising, but again - can move later.